src/reference/02-DetailTopics/02-Configuration/15-In-Process-Classloaders.md

```diff
@@ -0,0 +1,92 @@
+configuration, set
+
+```
+Test / classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.AllLibraryJars
+```
+
+Descriptions of the strategies follow.
+
+#### ScalaLibrary
+
+Prior to sbt 1.3.0, this was the only ClassLoader layering strategy. The top
+layer of this ClassLoader is able to load the scala standard library classes,
+the sbt test interface classes and the java library. This layer is cached
+between builds. The bottom layer of this classloader contains the rest of the
+project classpath which includes all of the library jar files as well as the
+project class files (or jars if `exportJars` is set). The bottom layer is never
+cached.
+
+Note that because the scala library is cached between runs, any shared global
+state created by the scala library is reused. In particular, the
+[global execution context](https://www.scala-lang.org/api/current/scala/concurrent/ExecutionContext\$.html)
+persists between runs. Avoid using the global execution context for tasks that
+run in process.
+
+The `ScalaLibrary` strategy is the default for all sbt 1.3.0 builds.
+
+#### AllLibraryJars
+
+Similar to `ScalaLibrary,` but adds an intermediate layer for the library
+dependencies of the project between the `ScalaLibrary` layer and the bottom
+project classpath layer. This strategy can significantly reduce the startup
+latency of run and test for projects with a large dependency classpath.
+
+Many projects will benefit in interactive sessions from switching to
+`AllLibraryJars`. The reason that it is not the default is because the caching
+of the dependency layer can cause errors if static fields are shared across
+runs. For example, some libraries use a global cache that is stored in a static
+field. If a project's `run` or `test` task depends on a particular initial
+state, they may fail on successive runs. Note that static global fields in the
+_project_ classpath do not cause this problem because the bottom layer is not
+cached between runs.
+
+The requirements for java serialization and deserialization to work are the same
+as that described in [Running Project Code][Running Project Code]. If, however,
+java serialization/deserialization of classes in the bottom layer is performed
+by classes in the dependency layer, then the dependency layer is invalidated
+between runs. When invalidation occurs, the performance of `AllLibraryJars` will
+be largely indistinguishable from `ScalaLibrary.`
+
+#### Flat
+
+This has only a single layer. It is essentially a URLClassLoader whose jars are
+the result of `fullClasspath.value.files`. Using this strategy should have
+similar characteristics as forking. The startup latency should be a bit lower
+than forking because it need not create a new jvm, but it has the same
+limitations with respect to shutdown hooks and threading as described in
+[Running Project Code][Running Project Code]. In most cases, `ScalaLibrary` or
+forking should be preferred.
```

Comment:
> In most cases, `ScalaLibrary` or forking should be preferred.

Since we've already promoted the goodness of other layering strategies, I don't think this sentence is necessary.
Instead, we should let the users know when `ClassLoaderLayeringStrategy.Flat` might come in handy.

For instance we can say:

> If in previous versions of sbt when forking was needed (due to the use of global ExecutionContext etc), consider trying `ClassLoaderLayeringStrategy.Flat`.